### PR TITLE
auto publish gh release on version tag

### PR DESCRIPTION
**Pull request by stefan-guggisberg**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
#306 

Thanks for contributing!

### CODE REVIEW

**Comment by stefan-guggisberg** [comment]
@tripodsan @trieloff the 'auto gh release' is triggered by a new version tag (`v1.2.3`). any idea how i can test this without messing with tags and releases?

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ jobs:
 
     steps:
     - setup
+    - checkout
```
Comment (tripodsan) [inline .circleci/config.yml]: checkout is already contain in the `setup` command

**Comment by tripodsan** [comment]
> the 'auto gh release' is triggered by a new version tag (`v1.2.3`). any idea how i can test this without messing with tags and releases?

maybe create tags manually by using a special tag, like: v99.3.1

File: .circleci/config.yml (hunk 1 above)
Comment (stefan-guggisberg) [reply .circleci/config.yml]: right, thanks.

**Comment by trieloff** [comment]
Or just mess with it, and then clean up later ;)

**Pull request merged**